pkg/eventmonitor/consumers/process.go

```diff
@@ -46,16 +46,28 @@ type event struct {
 var _ eventmonitor.EventConsumerHandler = &ProcessConsumer{}
 var _ eventmonitor.EventConsumer = &ProcessConsumer{}
 
+// ProcessConsumerOptions holds the options that can be passed to NewProcessConsumer
+type ProcessConsumerOptions struct {
+	// ID is the ID of the consumer that will be used to identify it with the event data stream
+	ID string
+
+	// ChanSize is the size of the channel that the event monitor will use to send events to this handler
+	ChanSize int
+
+	// ListenToForkEvents indicates if this consumer should listen to and handle fork events
+	ListenToForkEvents bool
```

Comment:
let's remove this explicit field, i don't see a reason to treat fork events differently than exec and exit events.

you can use [EventType](https://github.com/DataDog/datadog-agent/blob/main/pkg/security/secl/model/events.go#L9) as a parameter to the ProcessConsumer ctor to define interesting events.

Reply:
The problem with that approach is that then it looks like the consumer can accept any kind of event, and the check for whether the events are supported or not would be done in runtime. I think it's better if we limit this consumer as just a ready-to-use/happy-path consumer for process-monitor like consumers. Anything more complicated/customized than that, folks should implement their own.

Reply:
you can create a subset of process related events using an alias type if you want

Reply:
Done, let me know how it looks. I also reverted the options struct, as I think that forces users to think about the `eventTypes` parameter and not leave it as a default.